doc/specs/Configuration object for profiles.md

```diff
@@ -0,0 +1,94 @@
+---
+author: <Pankaj> <Bhojwani> <pabhojwa@microsoft.com>
+created on: <2020-11-20>
+last updated: <2020-11-20>
+issue id: <#8345>
+---
+
+# Configuration object for profiles
+
+## Abstract
+
+This spec outlines how we can support 'configuration objects' in our profiles, which
+will allow us to render differently depending on the state of the control. For example, a
+control can be rendered differently if its focused as compared to when its unfocused. Another
+example is that an elevated state control can be rendered differently as compared to a
+non-elevated one.
+
+## Inspiration
+
+Reference: [#3062](https://github.com/microsoft/terminal/issues/3062)
+
+Users want there to be a more visible indicator than the one we have currently for which
+pane is focused and which panes are unfocused. This change would grant us that feature.
+
+## Solution Design
+
+A new class in the `TerminalControl` namespace, called `CustomRenderConfig`, that will contain rendering
+parameters. `TermControl` and `TerminalCore` will use objects of this class to pass along rendering parameters
+to the renderer.
+
+Our `TerminalSettingsModel` will parse out that object from the settings json file and pipe it over to
+`TermControl`/`TerminalCore` to use. This will be done through `IControlSettings` and `ICoreSettings`, which
+is the way we already pipe over information that these interfaces need to know regarding rendering (such as
+`CursorStyle` and `FontFace`).
+
+### Allowed parameters

```

Comment:
So, when `"adminState"` gets added, we'll have a larger set of accepted parameters. For example...
- `name`, `tabTitle`, `suppressApplicationTitle` --> what is my title?
- `startingDirectory` and `commandline` --> startup-related settings

Should we just accept every profile setting, then silently ignore ones that don't apply?

Reply:
I'm thinking that at least for now, these additional state configurations should be entirely appearance-based, so we don't accept things like `commandline`, `startingDirectory` and probably nothing to do with the title either.